src/agent/advertising_proxy.cpp

```diff
@@ -187,6 +194,7 @@ void AdvertisingProxy::PublishServiceHandler(const char *aName, const char *aTyp
             }
             else
             {
+                update->mServiceNames.erase(nameAndType);
```

Comment:
```suggestion
                update->mPendingServiceNames.erase(nameAndType);
```
I suggest to rename to `mPendingServiceNames` to better reflect its usage.

Reply:
`mServiceNames` is already a field of `OutstandingUpdate` which should already indicate that the services and hosts are pending.

Reply:
SGTM